openmcyclus/DepleteReactor.py

```diff
@@ -609,23 +623,21 @@ def transmute(self):
         '''
         assemblies = self.core.pop_n(self.core.count)
         self.core.push_many(assemblies)
-        ss = str(len(assemblies)) + " assemblies"
+        # ss = str(len(assemblies)) + " assemblies"
         # self.record("TRANSMUTE", ss)
-        if self.check_existing_recipes(assemblies) == True:
-            for assembly in assemblies:
-                index = np.where(self.fresh_comps == assembly.comp())
-                assembly.transmute(self.spent_comps[index[0][0]])
-            return
         comp_list = [assembly.comp() for assembly in assemblies]
         material_ids, materials = self.deplete.update_materials(
             comp_list, self.materials)
         ind_op = od.IndependentOperator(
-            materials, self.micro_xs, str(
-                self.model_path + self.chain_file))
+                    materials,
+                    [np.array([self.flux])]*len(materials),
+                    [self.micro_xs]*len(materials),
```

Comment:
So we are assuming every material has the name flux profile and micro_xs?

Reply:
Yes. The current material handling only accounts for single-region cores, but with multiple batches. So unfortunately this isn't great for fast reactors, but it's a decent approximation for thermal reactors. I definitely have some ideas on how to expand this for multi-region cores that will be future work.